src/Symfony/Component/HttpClient/NativeHttpClient.php

```diff
@@ -142,6 +159,10 @@ public function request(string $method, string $url, array $options = []): Respo
                 $info['debug'] .= $info['request_header'];
                 unset($info['request_header']);
             } else {
+                if ($checkMaxDuration) {
```

Comment:
not needed with the suggested change above

Reply:
But I did it like this because $onProgress is called only on STREAM_NOTIFY_PROGRESS and STREAM_NOTIFY_CONNECT (it returns in the else). But I guess we wanna check the max duration on every tick in this callback.

Reply:
we don't care about the other notifications as they tell about things that are not related to network activity, so all is fine using $onProgress

Reply:
Interesting! In that case, it makes sense to use $onProgress directly.